plugins/geojson/src/main/java/com/mapbox/mapboxsdk/plugins/geojson/GeoJsonPlugin.java

```diff
@@ -0,0 +1,441 @@
+                            List<LatLng> latLngs = new ArrayList<>();
+                            JSONArray fullCoordinates = geometry.getJSONArray("coordinates");
+                            JSONArray coordinates = fullCoordinates.getJSONArray(0);
+                            for (int lc = 0; lc < coordinates.length(); lc++) {
+                                JSONArray coord = coordinates.getJSONArray(lc);
+                                LatLng latLng = new LatLng(coord.getDouble(1), coord.getDouble(0));
+                                latLngs.add(latLng);
+                                pointCount++;
+                                builder.include(latLng);
+                            }
+                            PolyData polygonPolyData = new PolyData();
+                            polygonPolyData.setPoints(latLngs);
+                            polygonPolyData.setType("Polygon");
+                            dataModel.addPolygon(polygonPolyData);
+                        } else {
+                            //TODO parse other type of Element like multiPolygon & etc
+                        }
+                    }
+                }
+            }
+        } catch (Exception exception) {
+            Log.e(EXCEPTION, "Exception Parse GeoJSON: " + exception.toString());
+        }
+        if (pointCount > 1)
+            dataModel.setBounds(builder.build());
+        return dataModel;
+    }
+
+    /**
+     * convert InputStream to String and pass to parseGeoJsonString function
+     *
+     * @param inputStream the input stream of GeoJson file
+     * @param listener    the instance of onLoadingGeoJsonListener
+     * @return DataModel that generated by parseGeoJsonString function
+     */
+    private DataModel parseGeoJsonInputStream(InputStream inputStream, OnLoadingGeoJsonListener listener) {
+        DataModel dataModel = null;
+        try {
+            BufferedReader rd = new BufferedReader(new InputStreamReader(inputStream, Charset.forName("UTF-8")));
+            StringBuilder sb = new StringBuilder();
+            int cp;
+            while ((cp = rd.read()) != -1) {
+                sb.append((char) cp);
+            }
+            inputStream.close();
+            if (!isJSONValid(sb.toString())) {
+                if (listener != null)
+                    listener.onLoadFailed(new IllegalStateException("GeoJson string is not valid"));
+            }
+            dataModel = parseGeoJsonString(sb.toString());
+        } catch (IOException e) {
+            Log.e(EXCEPTION, "Exception InputStream To String: " + e.toString());
+        }
+        return dataModel;
+    }
+
+    /**
+     * Drawing element of {@link DataModel} on the map
+     *
+     * @param dataModel list of polylines, polygons and points
+     */
+
+    private void DrawOnMap(DataModel dataModel) {
+        if (dataModel != null) {
+            if (dataModel.getMarkers() != null) {
+                if (dataModel.getMarkers().size() > 0) {
```

Comment:
these if checks can be simplified, avoid constantly getMarkers multiple times, assign to a local var instead + use` List#isEmpty` over `List#size() >0`